### PR TITLE
[DEV-1005] Api documentation

### PR DESCRIPTION
**Pull request by Kolibri1990**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,84 @@
 # LOCK API
 
-API for LOCK.space international staking service
\ No newline at end of file
+API for LOCK.space international staking service
+
+# Documentation
```
Comment (davidleomay) [inline README.md:5]: Use `##` for subtitles

File: README.md [hunk 2]
```diff
@@ -1,3 +1,84 @@
...
+- [Staking](#staking)
+- [Deposit](#deposit)
+- [Withdrawal](#withdrawal)
+- [Analytics](#analytics)
+- [Voting](#voting)
+- [Integration Example](#integration-example)
+
+# Registration
+
+1. Contact [support](support@lock.space) to registrate your wallet name
```
Comment (davidleomay) [inline README.md:18]: Link doesn't work, use `[support](mailto:support@lock.space)`?

File: README.md [hunk 3]
```diff
@@ -1,3 +1,84 @@
...
+2. Create an address on selected blockchain (e.g. DeFiChain)
+3. Get and sign message from sign-message endpoint: https://dev.api.lock.space/swagger/#/Authentication/AuthController_getSignMessage
```
Comment (davidleomay) [inline README.md:20]: Get the sign message from sign-message endpoint and sign it with the corresponding private key

File: README.md [hunk 4]
```diff
@@ -1,3 +1,84 @@
...
+4. Register the user with the sign-up endpoint: https://dev.api.lock.space/swagger/#/Authentication/AuthController_signUp
+5. Now you can get your access token (with address & signature) with the sign-in endpoint: https://dev.api.lock.space/swagger/#/Authentication/AuthController_signIn
```
Comment (davidleomay) [inline README.md:22]: I would hide the Swagger URLS (everywhere), e.g.
`Now you can get your access token (with address & signature) with the (sign-in endpoint)[https://dev.api.lock.space/swagger/#/Authentication/AuthController_signIn]`

File: README.md [hunk 5]
```diff
@@ -1,3 +1,84 @@
...
+
+# KYC
+
+Before the user can stake he has to proceed a KYC.
+
+1.  Trigger the kyc endpoint for KYC registration: https://dev.api.lock.space/swagger/#/KYC/KycController_startKyc
```
Comment (davidleomay) [inline README.md:28]: Trigger the **KYC** endpoint to start the KYC process

File: README.md [hunk 6]
```diff
@@ -1,3 +1,84 @@
...
+
+2.  User has to do KYC
+
+        a) If the address is already registered and the user processed the KYC process at DFX.swiss you can transfer the KYC data with the transfer endpoint at DFX API:
```
Comment (davidleomay) [inline README.md:32]: If the user is already verified (KYC completed) at DFX.swiss with the same address you can transfer the KYC data with the transfer endpoint at DFX API (use wallet name `LOCK.space`):

File: README.md [hunk 7]
```diff
@@ -1,3 +1,84 @@
...
+
+    https://api.dfx.swiss/swagger/#/auth/AuthController_signIn
+    https://api.dfx.swiss/swagger/#/kyc/KycController_transferKycData
+
+        b) Get and open the KYC Link which you can get from kyc endpoint after register KYC or from user endpoint ("kycLink"):
```
Comment (davidleomay) [inline README.md:37]: If not, the user can do KYC at LOCK. Get and open the KYC Link which you can get from KYC or user endpoint:

File: README.md [hunk 8]
```diff
@@ -1,3 +1,84 @@
...
+
+    https://dev.api.lock.space/swagger/#/User/UserController_getUser
```
Comment (davidleomay) [inline README.md:39]: image
This looks weird 🤔

File: README.md [hunk 9]
```diff
@@ -1,3 +1,84 @@
...
+
+3.  Check with the user endpoint if KYC is finished (kycStatus = 'Light'):
```
Comment (davidleomay) [inline README.md:41]: Check with the user endpoint if KYC is finished (KYC status `Light` or `Full`):

File: README.md [hunk 10]
```diff
@@ -1,3 +1,84 @@
...
+    https://dev.api.lock.space/swagger/#/User/UserController_getUser
+
+# Staking
+
+- Get the staking deposit address, id, information about fee, balance, min deposit, pending deposits, pending withdrawals etc. with the staking endpoint: https://dev.api.lock.space/swagger/#/Staking/StakingController_getStaking
+
+# Deposit
```
Comment (davidleomay) [inline README.md:48]: Use subsubtitle (###) for Deposit + Withdrawal

File: README.md (hunk 4 above)
Comment (Kolibri1990) [reply README.md:22]: Good point. I think you meant:

`Now you can get your access token (with address & signature) with the [sign-in endpoint](https://dev.api.lock.space/swagger/#/Authentication/AuthController_signIn)`

Comment (davidleomay) [reply README.md]: Ehh yes, true 😉

**Review by davidleomay — APPROVED** [review]

**Pull request merged**